corehq/apps/repeaters/models.py

```diff
@@ -284,14 +285,17 @@ def get_auth(self):
             return HTTPDigestAuth(self.username, self.password)
         return None
 
+    def send_request(self, payload, url, headers, auth):
+        return simple_post(payload, url, headers=headers, timeout=POST_TIMEOUT, auth=auth)
+
     def fire_for_record(self, repeat_record):
         headers = self.get_headers(repeat_record)
         auth = self.get_auth()
         payload = self.get_payload(repeat_record)
         url = self.get_url(repeat_record)
 
         try:
-            response = simple_post(payload, url, headers=headers, timeout=POST_TIMEOUT, auth=auth)
+            response = self.send_request(payload, url, headers, auth)
```

Comment:
to your question about whether to also pull out the other params: this is probably fine for the time being. My thinking was basically that since depending on what you actually do in `send_request`, you'll use different information and since it's all determined off of `self` and `repeat_record` we could just let each `send_request` gather its own headers, auth, etc as necessary. (For example, when you override it below, you don't use all the params.) That said, it's a tad tricky because for example `self.get_payload(repeat_record)` might raise an one of the exceptions handled in the except block below (it probably doesn't but I haven't thought too hard about it), so putting it inside would theoretically change the behavior in a way we don't want, etc.

I'm fine leaving it the way you have it now, and I'll likely be revisiting it in a later refactor down the line anyway.